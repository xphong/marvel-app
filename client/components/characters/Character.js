import PropTypes from 'prop-types';
import React, { Component } from 'react';

export default class Character extends Component {
  constructor(props){
    super(props);
  }

  render() {
    const {character} = this.props;

    return (
      <div className='ui card fadeIn-animation'>
        <div className='image'>
            <img src={character.image} />
        </div>
        <div className='content'>
            <div className='header'>{character.name}</div>
            <div className='description'>
              {character.description}
            </div>
          </div>
          <div className='extra content'>
            <span className='right floated'>
              <a target="_blank" href={character.url}>
                <button className='ui icon purple tiny button'>
                    More Info
                </button>
              </a>
            </span>
          </div>
      </div>
    );
  }
}

Character.propTypes = {
  character: PropTypes.object.isRequired
};