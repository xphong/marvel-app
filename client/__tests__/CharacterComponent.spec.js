import React from 'react';

import Character from '../components/characters/Character';

describe('Character Component', () => {
  it('should render Character', () => {
    const props = {
      character: {
        name: 'Hawkeye',
        description: 'poop'
      }
    };

    const wrapper = shallow(<Character {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
