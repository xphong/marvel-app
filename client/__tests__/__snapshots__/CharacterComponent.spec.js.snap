// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component should render Character 1`] = `
<div
  className="ui card fadeIn-animation"
>
  <div
    className="image"
  >
    <img
      alt="Image of Hawkeye"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Hawkeye
    </div>
    <div
      className="description"
    >
      poop
    </div>
  </div>
  <div
    className="extra content"
  >
    <span
      className="right floated"
    >
      <a
        target="_blank"
      >
        <button
          className="ui icon purple tiny button"
        >
          More Info
        </button>
      </a>
    </span>
  </div>
</div>
`;
